app/data_model/answer_store.py

```diff
@@ -168,25 +160,25 @@ def escaped(self):
             escaped.append(answer)
         return self.__class__(existing_answers=escaped)
 
-    def filter_by_location(self, location):
+    def filter_by_location(self, schema, location):
         """
         Find all answers in the answer store for a given location
 
+        :param schema: The QuestionnareSchema to lookup the answers for the location
         :param location: The location to filter results by
         return: Return a list of answers which match the location
         """
-        return self.filter(group_id=location.group_id, block_id=location.block_id, group_instance=location.
-                           group_instance)
+        answer_ids = schema.get_answer_ids_for_block(location.block_id)
 
-    def filter(self, group_id=None, block_id=None, answer_id=None, group_instance=None, answer_instance=None,
+        return self.filter(answer_ids=answer_ids, group_instance=location.group_instance)
+
+    def filter(self, answer_ids=None, group_instance=None, answer_instance=None,
```

Comment:
Should answer_ids now be required? It doesn't make much sense to delete all answers for a given group or answer instance.